conjur_api/providers/authn_authentication_strategy.py

```diff
@@ -40,15 +49,22 @@ async def login(self, connection_info: ConjurConnectionInfo, ssl_verification_da
         creds.api_key = await self._send_login_request(ssl_verification_data, connection_info, creds)
         return creds.api_key
 
-    async def authenticate(self, connection_info, ssl_verification_data) -> str:
+    async def authenticate(self, connection_info, ssl_verification_data) -> tuple[str, datetime]:
         """
         Authenticate uses the api_key (retrieved in `login()`) to fetch a short-lived conjur_api token that
         for a limited time will allow you to interact fully with the Conjur vault.
         """
         logging.debug("Authenticating to %s...", connection_info.conjur_url)
         creds = self._retrieve_credential_data(connection_info.conjur_url)
+
+        # If the credential provider already has a valid API token, return it
+        if self._is_authenticated(creds):
+            return str(creds.api_token), cast(datetime, creds.api_token_expiration)
```

Comment:
creds contains `Optional[str]` and `Optional[datetime]` but this function returns `str` and `datetime`.
Since we know in this `if` that these values are not `None` we need to cast them into the types that this function expects to return